nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/Notify.java

```diff
@@ -0,0 +1,191 @@
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.processors.standard.util.FlowFileAttributesSerializer;
+
+@EventDriven
+@SupportsBatching
+@Tags({"map", "cache", "notify", "distributed", "signal", "release"})
+@InputRequirement(Requirement.INPUT_REQUIRED)
+@CapabilityDescription("Caches a release signal identifier in the distributed cache, optionally along with "
+        + "the FlowFile's attributes.  Any flow files held at a corresponding Wait processor will be "
+        + "released once this signal in the cache is discovered.")
+@SeeAlso(classNames = {"org.apache.nifi.distributed.cache.client.DistributedMapCacheClientService", "org.apache.nifi.distributed.cache.server.map.DistributedMapCacheServer",
+        "org.apache.nifi.processors.standard.Wait"})
+public class Notify extends AbstractProcessor {
+
+    // Identifies the distributed map cache client
+    public static final PropertyDescriptor DISTRIBUTED_CACHE_SERVICE = new PropertyDescriptor.Builder()
+        .name("Distributed Cache Service")
+        .description("The Controller Service that is used to check for release signals from a corresponding Notify processor")
+        .required(true)
+        .identifiesControllerService(DistributedMapCacheClient.class)
+        .build();
+
+    // Selects the FlowFile attribute or expression, whose value is used as cache key
+    public static final PropertyDescriptor RELEASE_SIGNAL_IDENTIFIER = new PropertyDescriptor.Builder()
+        .name("Release Signal Identifier")
+        .description("A value, or the results of an Attribute Expression Language statement, which will " +
+            "be evaluated against a FlowFile in order to determine the release signal cache key")
+        .required(true)
+        .addValidator(StandardValidators.createAttributeExpressionLanguageValidator(ResultType.STRING, true))
+        .expressionLanguageSupported(true)
+        .build();
+
+    // Specifies an optional regex used to identify which attributes to cache
+    public static final PropertyDescriptor ATTRIBUTE_CACHE_REGEX = new PropertyDescriptor.Builder()
+        .name("Attribute Cache Regex")
+        .description("Any attributes whose names match this regex will be stored in the distributed cache to be "
+                + "copied to any FlowFiles released from a corresponding Wait processor.  Note that the "
+                + "uuid attribute will not be cached regardless of this value.  If blank, no attributes "
+                + "will be cached.")
+        .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+        .expressionLanguageSupported(false)
+        .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+        .name("success")
+        .description("A FlowFile with a matching release signal in the cache will be routed to this relationship")
```

Comment:
Technically it would be: "The first FlowFile with a matching release signal in the cache...", since Wait removes the cache entry once it has successfully transferred the flow file to success.  

Do you think there should be a use case where the cache value is not removed?  If so, we could add a property like "Remove Successful Signals?"